components/server/ee/src/workspace/workspace-factory.ts

```diff
@@ -89,18 +89,29 @@ export class WorkspaceFactoryEE extends WorkspaceFactory {
             const { project, branch } = context;
 
             const commitContext: CommitContext = context.actual;
-            const existingPWS = await this.db
-                .trace({ span })
-                .findPrebuiltWorkspaceByCommit(
-                    commitContext.repository.cloneUrl,
-                    CommitContext.computeHash(commitContext),
-                );
-            if (existingPWS) {
+
+            const asserNoPrebuildIsRunningForSameCommit = async () => {
```

Comment:
Would be nice to have a comment here (or on usage-sites below) that summarizes _why_ we're calling it repeatedly (because we're being pragmatic).

Reply:
done